include/nanobind/nb_class.h

```diff
@@ -386,6 +391,20 @@ class class_ : public object {
             }
         }
 
+        if constexpr (detail::has_shared_from_this_v<T>) {
+            d.flags |= (uint32_t) detail::type_flags::has_shared_from_this;
+            d.keep_shared_from_this_alive = [](PyObject *self) noexcept {
+                if (auto sp = inst_ptr<T>(self)->weak_from_this().lock()) {
```

Comment:
Could you add 1-line comment here explaining what this does? (I never used the `weak_from_this().lock()` construction).